Sources/SwiftParser/Specifiers.swift

```diff
@@ -415,7 +415,108 @@ extension RawAccessorEffectSpecifiersSyntax: RawEffectSpecifiersTrait {
       }
     }
   }
+}
+
+// This conformance exists solely to parse misplaced specifiers in parseFunctionReturnClause()
+extension RawDeinitEffectSpecifiersSyntax: RawEffectSpecifiersTrait {
+  enum MisspelledAsyncSpecifiers: TokenSpecSet {
+    case await
+    case reasync
+
+    init?(lexeme: Lexer.Lexeme) {
+      switch PrepareForKeywordMatch(lexeme) {
+      case TokenSpec(.await, allowAtStartOfLine: false): self = .await
+      case TokenSpec(.reasync): self = .reasync
+      default: return nil
+      }
+    }
+
+    var spec: TokenSpec {
+      switch self {
+      case .await: return TokenSpec(.await, allowAtStartOfLine: false)
+      case .reasync: return .keyword(.reasync)
+      }
+    }
+  }
+
+  enum CorrectAsyncTokenKinds: TokenSpecSet {
+    case async
+
+    init?(lexeme: Lexer.Lexeme) {
+      switch PrepareForKeywordMatch(lexeme) {
+      case TokenSpec(.async): self = .async
+      default: return nil
+      }
+    }
+
+    var spec: TokenSpec {
+      switch self {
+      case .async: return .keyword(.async)
+      }
+    }
+  }
+
+  enum MisspelledThrowsTokenKinds: TokenSpecSet {
+    case `rethrows`
+    case `try`
+    case `throw`
+    case `throws`
+
+    init?(lexeme: Lexer.Lexeme) {
+      switch PrepareForKeywordMatch(lexeme) {
+      case TokenSpec(.rethrows): self = .rethrows
+      case TokenSpec(.try, allowAtStartOfLine: false): self = .try
+      case TokenSpec(.throw, allowAtStartOfLine: false): self = .throw
+      case TokenSpec(.throws): self = .throws
+      default: return nil
+      }
+    }
+
+    var spec: TokenSpec {
+      switch self {
+      case .rethrows: return .keyword(.rethrows)
+      case .try: return TokenSpec(.try, allowAtStartOfLine: false)
+      case .throw: return TokenSpec(.throw, allowAtStartOfLine: false)
+      case .throws: return .keyword(.throws)
+      }
+    }
+  }
+
+  enum CorrectThrowsTokenKinds: TokenSpecSet {
+    // Uninhabited
 
+    init?(lexeme: Lexer.Lexeme) {
+      return nil
+    }
+
+    var spec: TokenSpec {
+      switch self {
+      }
+    }
+  }
+
+  var unexpectedBetweenAsyncSpecifierAndThrowsSpecifier: RawUnexpectedNodesSyntax? { unexpectedAfterAsyncSpecifier }
+  var throwsSpecifier: RawTokenSyntax? { nil }
+  var unexpectedAfterThrowsSpecifier: RawUnexpectedNodesSyntax? { nil }
+
+  init(
+    _ unexpectedBeforeAsyncSpecifier: RawUnexpectedNodesSyntax?,
+    asyncSpecifier: RawTokenSyntax?,
+    _ unexpectedBetweenAsyncSpecifierAndThrowsSpecifier: RawUnexpectedNodesSyntax?,
+    throwsSpecifier: RawTokenSyntax?,
+    _ unexpectedAfterThrowsSpecifier: RawUnexpectedNodesSyntax?,
+    arena: __shared SwiftSyntax.SyntaxArena
+  ) {
+    // Inserted missing throws is discarded
+    assert(throwsSpecifier?.isMissing ?? true)
```

Comment:
Could you change this to a `precondition`?

https://github.com/apple/swift-syntax/blob/8a759449c646c72b9392b151236204a5fad03831/Sources/SwiftSyntax/Assert.swift#L13-L22

Reply:
Done

Reply:
Did you forget to push? Still appears to be an `assert` to me.